### DISCUSSION TITLE
data fetch에 관한 이야기

### DISCUSSION THREAD

**Discussion by cyberjam** [LifeIsAlone/public-home#57] [General]
스프레드 시트를 업데이트 했는데 배포된 지도에 나타나지 않았습니다. 버셀에 가서 redeploy를 하니 그제서야 업데이트 되더군요.

접속시 계속 스프레드시트에서 불러오는줄만 알았습니다. 그래서 스프레드시트 접속 수도 함께 고민했었죠. json으로 어딘가 저장되게 한다면 접속 수를 줄일수 있겠다고도 했구요.

어쩌면 잘못 생각했을수 있겠어요.

스프레드시트를 getStaticProps로 가져왔거든요. 그러면 deploy당시 데이터를 정적으로 한번만 가져온다는 소리로 들립니다.
https://coffeeandcakeandnewjeong.tistory.com/97

**Comment by sangjuun-park** [comment]
그렇네요, 매번 호출할 줄 알았는데, product로 빌드시 getStaticProps는 데이터도 가져오네요
.Next에 build 해서 확인 해봤습니다.
![image](https://user-images.githubusercontent.com/101157141/193439586-819f5218-cfab-4c14-8816-a9d597c36f2d.png)


더 나아가서 만약 데이터를 변경하면 반영이 안되는 거 잖아요? ( 반영할려면 다시 재 배포를 수기로 해야하는 거니까요)
저번에 쓰신 validate를 Props에 넣으면 되는 것 같습니다.

참고자료: https://www.youtube.com/watch?v=khcRPut9DMA
